machina/management/commands/send_notifications.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+from django.utils import translation
+
+from machina.apps.forum_member.utils import send_notification
+from machina.core.db.models import get_model
+
+
+Post = get_model('forum_conversation', 'Post')
+
+
+class Command(BaseCommand):
+    help = 'Send email to users that have turned on notifications for subscribed topics.'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--interval',
+            dest='interval',
+            default=300,  # 5 minutes
+            type=int,
+            help=('Subscribers on posts created within this interval (in seconds) will '
+                  'receive notifications on email.'),
+        )
+
+    def get_recent_posts(self, interval):
+        """
+        Return new posts that was created within x seconds ago.
+        """
+        time_ago = timezone.now() - timedelta(seconds=interval)
+        return Post.objects.filter(created__gt=time_ago)
+
+    def handle(self, *args, **options):
```

Comment:
A command is surely the way to go in many cases but some projects could also rely on some asynchronous workers to run such tasks. For example an asynchronous task could be triggered when a new post is created in order to send the related notifications. My point is that this command embeds some logic that could be used by such tasks if needed. Could you refactor it so that your command calls a function whose purpose is to send these e-mails? If you do this, it'd be easier to use your implementation with Celery (or channels) to send these notifications without having to copy paste these lines. 😃